nowcasting_dataset/dataset/split/day.py

```diff
@@ -0,0 +1,105 @@
+from typing import Union, List, Tuple
+
+import random
+import numpy as np
+import pandas as pd
+
+
+def split_day(
+    datetimes: Union[List[pd.Timestamp], pd.DatetimeIndex],
+    train_test_validation_split: Tuple[int] = (3, 1, 1),
+) -> (List[pd.Timestamp], List[pd.Timestamp], List[pd.Timestamp]):
+    """
+    Split the data by the day.
+
+    If the split is (3,1,1) then
+    train data will have all days of the year that are module 5 remainder 0,1 or 2
+    validation data will have all days of the year that are module 5 remainder 3
+    test data will have all days of the year that are module 5 remainder 4
+
+    Args:
+        datetimes: list of datetimes
+        train_test_validation_split: how the split is made
+
+    Returns: train, validation and test datetimes
+
+    """
+
+    total_weights = sum(train_test_validation_split)
+    cum_weights = np.cumsum(train_test_validation_split)
+
+    # make which day indexes go i.e if the split is [3,1,1] then the
+    # - train_ indexes = [0,1,2]
+    # - validation_indexes = [3]
+    # - test_indexes = [4]
+    train_indexes = [i for i in range(total_weights) if i < cum_weights[0]]
+    validation_indexes = [
+        i for i in range(total_weights) if (i >= cum_weights[0]) & (i < cum_weights[1])
+    ]
+    test_indexes = [i for i in range(total_weights) if (i >= cum_weights[1]) & (i < cum_weights[2])]
+
+    # select the datetimes
+    train = [
+        datetime for datetime in datetimes if datetime.dayofyear % total_weights in train_indexes
+    ]
+    validation = [
+        datetime
+        for datetime in datetimes
+        if datetime.dayofyear % total_weights in validation_indexes
+    ]
+    test = [
+        datetime for datetime in datetimes if datetime.dayofyear % total_weights in test_indexes
+    ]
+
+    return train, validation, test
+
+
+def split_day_random(
+    datetimes: Union[List[pd.Timestamp], pd.DatetimeIndex],
+    train_test_validation_split: Tuple[int] = (3, 1, 1),
+) -> (List[pd.Timestamp], List[pd.Timestamp], List[pd.Timestamp]):
+    """
+    Split the data by the day. Take random days
+
+    If the split is (3,1,1) then
+    train data will have 60% of the days
+    validation data will have 20% of the days
+    test data will have have 20% of the days
+
+    Args:
+        datetimes: list of datetimes
+        train_test_validation_split: how the split is made
+
+    Returns: train, validation and test datetimes
+
+    """
+
+    total_weights = sum(train_test_validation_split)
+    cum_weights = np.cumsum(train_test_validation_split)
+
+    if type(datetimes) is pd.DatetimeIndex:
```

Comment:
I could be wrong but I don't _think_ these two lines are necessary.  `pd.DatetimeIndex(datetimes)` (on line 84) will do the right thing, no matter if `datetimes` is a `List[pd.Timestamp]` or a `pd.DatetimeIndex`.

In fact, I'd recommend replacing lines 80 and 81 with: `datetimes = pd.DatetimeIndex(datetimes)` so the rest of the function knows that `datetimes` will definitely be a `DatetimeIndex`.

Reply:
Yea, in fact Ill push this higher up into the split_data function

Reply:
yea, ive moved this now